skbio/diversity/alpha/_base.py

```diff
@@ -848,6 +850,51 @@ def osd(counts):
     return observed_otus(counts), singles(counts), doubles(counts)
 
 
+@experimental(as_of="0.4.0-dev")
+def pielou_e(counts):
+    r"""Calculate Pielou's Evenness index J'.
+
+    Pielou's Evenness is defined as:
+
+    .. math::
+
+       J' = \frac{(H)}{\ln(S)}
+
+    where :math:`H` is the Shannon-Wiener entropy of counts and :math:`S` is
+    the number of OTUs in the sample.
+
+    Parameters
+    ----------
+    counts : 1-D array_like, int
+        Vector of counts.
+
+    Returns
+    -------
+    double
+        Pielou's Evenness.
+
+    See Also
+    --------
+    shannon
+    heip_e
+
+    Notes
+    -----
+    The implementation here is based on the description given by Carlo Heip
+    [1]_ and is also described on Wikipedia [2]_. It is similar to Heip's
+    evenness [1]_.
+
+    References
+    ----------
+    .. [1] Heip, C. 1974. A new index measuring evenness. J. Mar. Biol. Ass.
+       UK., 54, 555-557.
+    .. [2] https://en.wikipedia.org/wiki/Species_evenness
+
+    """
+    counts = _validate_counts_vector(counts)
+    return shannon(counts) / np.log(observed_otus(counts))
```

Comment:
I'll be using this site for external ref: http://ww2.mdsg.umd.edu/interactive_lessons/biofilm/diverse.htm#3

I'm now calling `shannon(base=np.e)`, which works fine. The choice of a default is beyond my pay grade and both will work fine in the code.

Open question: when only one species is in the community, what should we return for Pielou's evenness?
- Option one: return `0` because Shannon entropy is zero.
- Option two: return `nan` because we are dividing by log(1), which is zero.